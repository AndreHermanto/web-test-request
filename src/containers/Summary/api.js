/**
 * Submit the test request
 * @param {Object} data Test request data to be submitted.
 */
export function submitTestRequest(data) {
  return fetch(`${process.env.REACT_APP_BASE_URL}/test-requests`, {
    method: 'POST',
    headers: {
      'Content-Type': 'application/json'
    },
    body: JSON.stringify(data)
  });
}
