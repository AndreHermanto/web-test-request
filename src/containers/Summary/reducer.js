import validator from './../../components/validator';

export function initData(testRequest) {
  var state = {
    form: {
      testRequest,
      signature:false
    },
    validationRule: {
      signature: 'signatureTrue',
    },
    validated: false,
    submitStatus:''
  };
  // This validates the data in the initial state.
  state.validation = {};
  for (var field in state.validationRule) {
    if(field) {
      state.validation[field] = validator(state.form[field], state.validationRule[field]);
    }
  }
  
  return Object.assign({}, state);
}

export function setSubmitStatusData(state, value)
{
  return Object.assign({}, state, {
    submitStatus: value
  });
}

/**
 * This sets the form data upon onChange.
 * @param {Object} state Targeted state to be changed.
 * @param {Object} target Target object captured from UI event change.
 */
export function setSignatureData(state, target) {
  var value = target.checked;
  
  var formStateChild = Object.assign({}, state.form, {
    [target.name]: value
  });
  
  var validationStateChild = Object.assign({}, state.validation, {
    [target.name]: validator(value, state.validationRule[target.name])
  });

  return Object.assign({}, state, {
    form: formStateChild,
    validation: validationStateChild
  });
}
/**
 * Set "validated" state to true - identifying the confirm button is clicked and validation processed.
 * @param {Object} state Targeted state to be changed.
 */
export function validatedToTrue(state) {
  return Object.assign({}, state, {
    validated: true
  });
}

