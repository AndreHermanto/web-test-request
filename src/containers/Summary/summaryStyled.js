import styled from 'styled-components';
import { Label } from 'react-bootstrap';

export const SummaryBox = styled.div`
  border: 1px solid rgb(227, 231, 241);
  width: 100%;
  padding: 20px;
  border-bottom: 0px solid transparent;
`;
export const LastBox = styled.div`
  border: 1px solid rgb(227, 231, 241);
  width: 100%;
  padding: 20px;
  margin-bottom:21px;
`;

export const SummaryHeading = styled.h3`
  margin-top: 10px !important;
  font-size: 22px !important;
`;

export const SectionHeading = styled.h4`
  margin-top: 10px !important;
`;

export const SummaryBreakLine = styled.div`
  border-bottom: 1px solid rgb(227, 231, 241);
  margin-bottom: 15px;
`;

export const SummaryTitle = styled.span`
  margin-bottom: 10px;
  font-weight: 200;
`;

export const SummaryDetails = styled.p`
  margin-bottom: 10px;
  font-weight: 200;
  line-height: 1;
`;

export const Gene = styled(Label)`
  display: inline-block !important;
  font-weight: 300 !important;
  background-color: #00a6b6 !important;
  padding: 5px 10px 5px 10px !important;
  margin-left:5px;
`;
export const GeneLabel = styled(Label)`
  margin-right: 4px;
  display: inline-block !important;
  font-weight: 300 !important;
`;
export const SummaryNotes = styled.div`
  background-color: #fafafa;
  padding: 1.3em;
  border-bottom: 1px solid rgb(227, 231, 241);
  margin-bottom: 10px;
`;

export const Tag = styled(Label)`
  margin-left: 10px;
  position: absolute;
  line-height: normal !important;
  font-size: 12px !important;
  margin-top: 9px;
`;

