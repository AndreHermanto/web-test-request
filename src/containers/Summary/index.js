import React, { Component } from 'react';
import {
} from 'react-bootstrap';
import { submitTestRequest } from './api';
import { 
  PageHeading,
  FormButton
} from './../../components/SharedStyle';

/**
 * Summary - UI for ordering type of tests, selecting disorder and related genes for testing.
 */
class Summary extends Component {
  constructor(props) {
    super(props);
    this.handleBack = this.handleBack.bind(this);
    this.handleSubmit = this.handleSubmit.bind(this);
    this.state = { form: props.route.data || {} }
  }
  
  handleBack() {
    this.props.router.push('/step5');
  }
  
  handleSubmit() {
    return submitTestRequest(this.state.form)
      .then((response) => {
        if(!response.ok) {
          alert('Submit fail');
          return false;
        } else {
          alert('Submit success');
          return true;
        }
      })
  }

  render() {
    return (
      <div>
        <PageHeading>Summary</PageHeading>
      
        <FormButton 
          bsStyle="warning" 
          onClick={this.handleBack}
          label="Back"
        >
          Back
        </FormButton> 
      
        <FormButton 
          bsStyle="success" 
          type="submit" 
          onClick={this.handleSubmit}
        >
          Submit
        </FormButton> 
      </div>
    );
  }
}

export default Summary;