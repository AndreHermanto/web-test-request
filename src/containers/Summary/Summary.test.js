import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import TestUtils from 'react-addons-test-utils';
import FetchMock from 'fetch-mock';
import Summary from './index';

describe('Summary: index', () => {
    const testData = {
    test: {
    id:'7052d137-a166-48b0-a52e-e05a167bd176',
    label: 'Consequatur adipisci modi laudantium tenetur ea exercitationem id',
    genes:["GZOW","KIHG","DRCO","DWBY","ZMZY","BMBP","OLZO","PZOA"]
    },
    genes:["GZOW","KIHG","DRCO","DWBY","ZMZY","BMBP","OLZO","PZOA"]
  }
  const patientData = {
    "lastName":"abcabc",
    "firstName":"Ngabcuyen",
    "dob":"1-February-1917",
    "medicalRecordNo":"123123",
    "gender":"Male",
    "genderOther":"",
    "ethnicity":"Asian",
    "deceased":false,
    "sampleSource":"",
    "consent":false,
    "email":"abc@abc.gmail.com"
  }
  const clinicianDetails = {
    "providerNumber":"as","medicalSpecialty":"as","firstName":"John",
    "lastName":"Doe","organisation":"as","phone":"as","email":"as","fax":"as",
    "copyToHCP":[{"additionalFirstName":"asdf","additionalLastName":"asdf","additionalOrganisation":"adsf","additionalEmail":"asfd"},
    {"additionalFirstName":"asdf","additionalLastName":"asdf","additionalOrganisation":"adsf","additionalEmail":"asdf"}],
    "copy":false
  }
  const clinicalData = {
    clinicalInfo: "Dissuade ecstatic and properly saw entirely sir why laughter endeavor. In on my jointure horrible margaret suitable he followed speedily. Indeed vanity excuse or mr lovers of on. By offer scale an stuff. Blush be sorry no sight. Sang lose of hour then he left find",
    consangunity: false,
    familyHistory: "kasdfbljkbfalhfasvfkjabskldfbkjsdfbkdashfvdjhfhb;kfjv;hfv;fjbafhb;akfd",
    relevantInvestigation:"asdkfhgasdlfhabsdlfjhalsfdjh"
  }

  const billingInfo = {
    payer:"abc abc",
    phone:"+61412334477",
    email:"abc@gmail.com",
    givenName:"abc",
    lastName:"abc",
    billOption:"Private"
  }
  
  var props = {
    route: {
      onChange: jest.fn(),
      onEdit: jest.fn(),
      data: {
        OrderTest: testData,
        PatientDetails: patientData,
        ClinicalInfo: {clinicalInfo:'abacbascs'},
        ClinicianDetails: clinicianDetails,
        BillingInfo: billingInfo
      }
    },
    router:['/summary']
  };

  
  test('renders without crashing - initial state with empty geneList', () => {
    const page = renderer.create(React.createElement(Summary, props)).toJSON();
    expect(page).toMatchSnapshot();
  });
  
  test('handleBack works', () =>  {
    var page = TestUtils.renderIntoDocument(React.createElement(Summary, props));                                         
    page.handleBack();
    expect(page.props.router.pop()).toEqual('/step6')
  });
  
  test('handleSubmit works', async () => {
    var page = TestUtils.renderIntoDocument(React.createElement(Summary, props));
    FetchMock.mock('*', 200);
    
    await page.handleSubmit().then((response) => {
      expect(response).toEqual(true)
    });
    
    FetchMock.restore();  
    FetchMock.mock('*', 404);
    
    await page.handleSubmit().then((response) => {
      expect(response).toEqual(false)
    });
    
    FetchMock.restore();
  });
});
