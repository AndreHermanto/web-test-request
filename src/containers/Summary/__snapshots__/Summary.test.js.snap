exports[`Summary: index renders without crashing - initial state with empty geneList 1`] = `
<div>
  <h3
    className="fAmrmf">
    Summary
  </h3>
  <button
    className="iMtXso btn btn-warning"
    disabled={false}
    label="Back"
    onClick={[Function]}
    type="button">
    Back
  </button>
  <button
    className="iMtXso btn btn-success"
    disabled={false}
    onClick={[Function]}
    type="submit">
    Submit
  </button>
</div>
`;
